lesson21/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+    {dataOrder:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {dataOrder:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortByUsersTableColumn = {
+    "id" : "ID"
+}
+
+const addSortButtons = (userKey,userValue,th) => {
+    for(const value of Object.values(sortByUsersTableColumn)){
+        userValue === value && createSortButtons(userKey,th) ;
+    }
+}
+
+const createSortButtons = (columnKey,th) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtons){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-${columnKey}SortButton`,
+                'data-order':button.dataOrder
+            }
+        });
+
+        button.dataOrder !== "both" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+    th.appendChild(sortButtonsBox).appendChild(fragment);
+}
+
+const toggleHiddenClassSortButton = (sortButtons,index) => {
+    sortButtons[index].classList.add("hidden");
+    (index === sortButtons.length -1) ? index = 0 : index++;
+    sortButtons[index].classList.remove("hidden");
+}
+
+const upDateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (key, usersData, dataOrder) => {
+    const cloneUsersData = [...usersData];
+
+    const sortButtonsFunc = {
+        "both": () => {
+            cloneUsersData.sort((a,b) => a[key] - b[key]);
+            upDateTableBody(cloneUsersData);
+        },
+        "asc" : () => {
+            cloneUsersData.sort((a,b) => b[key] - a[key]);
+            upDateTableBody(cloneUsersData);
+        },
+        "desc": () => {
+            upDateTableBody(usersData);
+        }
+    }
+
+    Object.keys(sortButtonsFunc).forEach((button) => dataOrder === button && sortButtonsFunc[button]() );
+}
+
+const sortTableBody = usersData => {
+    for(const key of Object.keys(sortByUsersTableColumn)){
+        const sortButtons = [...document.getElementsByClassName(`js-${key}SortButton`)];
+        const sortButtonsBox = document.getElementById(`js-${key}Buttons-Box`);
+        
+        let dataOrder;
+        sortButtonsBox.addEventListener("click", (e) => {
+            let index = sortButtons.indexOf(e.target);
+            dataOrder = e.target.getAttribute("data-order");
```

Comment:
Q
Is there a reason you need a `let`?

Reply:
As you pointed out, there was no need to use let, in the index.
I fixed.
https://github.com/nakagawamai/js-lesson/pull/23/commits/9c4eda9f668cea7d8cc5ea1dc9712095e0966201